src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -2337,5 +2349,48 @@ private IInstanceReferenceOperation CreateCollectionValuePlaceholderOperation(Bo
             bool isImplicit = placeholder.WasCompilerGenerated;
             return new InstanceReferenceOperation(referenceKind, _semanticModel, syntax, type, isImplicit);
         }
+
+        private ImmutableArray<IArgumentOperation> CreateDisposeArguments(MethodSymbol disposeMethod, TypeSymbol receiverType, SyntaxNode receiverSyntax, Binder binder)
+        {
+            // can't be an extension method for dispose
+            Debug.Assert(!disposeMethod.IsStatic);
+
+            if (disposeMethod.ParameterCount == 0)
+            {
+                return ImmutableArray<IArgumentOperation>.Empty;
+            }
+
+            var receiver = new BoundDisposableValuePlaceholder(receiverSyntax, receiverType);
+
+            var argumentsBuilder = ArrayBuilder<BoundExpression>.GetInstance(disposeMethod.ParameterCount);
+            ImmutableArray<int> argsToParams = default;
+            var expanded = disposeMethod.HasParamsParameter();
+
+            binder.BindDefaultArguments(
+                receiver.Syntax,
+                disposeMethod.Parameters,
+                argumentsBuilder,
+                argumentRefKindsBuilder: null,
+                ref argsToParams,
+                out BitVector defaultArguments,
+                expanded,
+                enableCallerInfo: true,
+                new DiagnosticBag());
+
+            var args = DeriveArguments(
+                            receiver,
```

Comment:
>receiver [](start = 28, length = 8)

It looks like this parameter is not used (at least in master). If that is the case, I think we should remove it and simplify consumers. For example, for this change we wouldn't need to create the receiver node and, therefore, wouldn't need to go through the trouble of figuring out its type in order to call ```CreateDisposeArguments```.

Reply:
Oh weird, good catch. There are a couple of unused arguments to that method. I've removed them and simplified the call sites to not need the type.

---
In reply to: [540155631](https://github.com/dotnet/roslyn/pull/49842#discussion_r540155631) [](ancestors = 540155631)